internal/core/util/src/main/java/io/quarkiverse/azure/core/util/AzureQuarkusIdentifier.java

```diff
@@ -13,6 +13,7 @@ public final class AzureQuarkusIdentifier {
     public static final String AZURE_QUARKUS_STORAGE_BLOB = "az-qk-storage-blob";
     public static final String AZURE_QUARKUS_KEY_VAULT_ASYNC_CLIENT = "az-qk-kv-secret-async";
     public static final String AZURE_QUARKUS_KEY_VAULT_SYNC_CLIENT = "az-qk-kv-secret-sync";
```

Comment:
I am not sure whether the identifier here will work, I remembered there is a length limitation here.

Reply:
The max length of [ClientOptions#applicationId](https://github.com/Azure/azure-sdk-for-java/blob/azure-core_1.23.1/sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java#L17)  is 24

@majguo check [this](https://dev.azure.com/SpringOnAzure/Spring%20on%20Azure/_wiki/wikis/spring-integration-private.wiki/264/Design-Collecting-artifact's-usage-count)

Reply:
I think we also need to have a design for this.

Reply:
You're right, the length of the app id can't be greater than 24. It's guarded by caller function `setApplicationId`:
![image](https://github.com/user-attachments/assets/22aac536-2fcc-490f-9b77-b6e7955b3d5c)

We also have a comment in class `AzureQuarkusIdentifier` , see https://github.com/quarkiverse/quarkus-azure-services/blob/main/internal/core/util/src/main/java/io/quarkiverse/azure/core/util/AzureQuarkusIdentifier.java#L11.

Reply:
> I think we also need to have a design for this.

Good idea. Simply said, the format of app id follows the pattern `az-qk-<extension-name>` with the limitation that the length can't be greater than 24. I also created a user story for tracking the request here: https://dev.azure.com/azure-redhat-projects/azure-redhat-projects/_workitems/edit/845.